src/main/java/net/dries007/tfc/world/chunkdata/RegionChunkDataGenerator.java

```diff
@@ -126,17 +128,20 @@ public ChunkData generate(ChunkData data)
 
         // The base rainfall and temperature layers, scaled down to chunk resolution
         LerpFloatLayer rainfallLayer = rainfallGridLayer.scaled(deltaX, deltaZ, dG);
+        LerpFloatLayer rainfallVarianceLayer = rainfallVarianceGridLayer.scaled(deltaX, deltaZ, dG);
+        LerpFloatLayer baseGroundwaterLayer = baseGroundwaterGridLayer.scaled(deltaX, deltaZ, dG);
         LerpFloatLayer temperatureLayer = temperatureGridLayer.scaled(deltaX, deltaZ, dG);
 
         // Calculate local influence of rivers - when they are wide enough (which should happen only with large rivers near shores),
-        // rivers will influence the rainfall of the nearby area, thus creating a bit of a humid area in the immediate vicinity of the river
+        // rivers will influence the groundwater of the nearby area, thus creating a bit of a humid area in the immediate vicinity of the river
+        // groundwater is used instead of rainfall in order to avoid strange rain effects
         //
         // The radius that the partition point will find rivers is approx ~5 grid distance (~500 blocks), so we should be fine to influence
         // on a smaller scale of ~100 blocks around the river
-        float rainfall00 = rainfallLayer.value00();
-        float rainfall01 = rainfallLayer.value01();
-        float rainfall10 = rainfallLayer.value10();
-        float rainfall11 = rainfallLayer.value11();
+        float groundwater00 = baseGroundwaterLayer.value00();
+        float groundwater01 = baseGroundwaterLayer.value01();
+        float groundwater10 = baseGroundwaterLayer.value10();
+        float groundwater11 = baseGroundwaterLayer.value11();
```

Comment:
Which means these could be initialized to all zeros
```suggestion
        float groundwater00 = 0;
        float groundwater01 = 0;
        float groundwater10 = 0;
        float groundwater11 = 0;
```